lib/engine/game/g_18_norway/game.rb

```diff
@@ -0,0 +1,131 @@
+        include Trains
+        include Phases
+
+        register_colors(red: '#d1232a',
+                        orange: '#f58121',
+                        black: '#110a0c',
+                        blue: '#025aaa',
+                        lightBlue: '#8dd7f6',
+                        yellow: '#ffe600',
+                        green: '#32763f',
+                        brightGreen: '#6ec037')
+
+        TRACK_RESTRICTION = :permissive
+        SELL_BUY_ORDER = :sell_buy
+        TILE_RESERVATION_BLOCKS_OTHERS = :always
+        CURRENCY_FORMAT_STR = '%skr'
+        EBUY_SELL_MORE_THAN_NEEDED = true
+        CAPITALIZATION = :incremental
+        MUST_BUY_TRAIN = :always
+        POOL_SHARE_DROP = :left_block
+        SELL_AFTER = :p_any_operate
+        SELL_MOVEMENT = :left_block
+        CERT_LIMIT_COUNTS_BANKRUPTED = true
+        HOME_TOKEN_TIMING = :float
+
+        BANK_CASH = 999_000
+
+        CERT_LIMIT = {
+          3 => { 0 => 12, 1 => 12, 2 => 12, 3 => 15, 4 => 15, 5 => 17, 6 => 17, 7 => 19, 8 => 19 },
+          4 => { 0 => 9, 1 => 9, 2 => 9, 3 => 11, 4 => 11, 5 => 13, 6 => 13, 7 => 15, 8 => 15 },
+          5 => { 0 => 7, 1 => 7, 2 => 7, 3 => 9, 4 => 9, 5 => 10, 6 => 10, 7 => 12, 8 => 12 },
+        }.freeze
+
+        STARTING_CASH = { 3 => 400, 4 => 300, 5 => 240 }.freeze
+
+        MARKET = [
+          %w[0c 10 20 30 40
+             50p 60p 70p 80p 90p 100p 112p 124p 137p 150p
+             165 180 195 220 245 270 300 330 365 400 440 480],
+           ].freeze
+
+        ASSIGNMENT_TOKENS = {
+          'MOUNTAIN_SMALL' => '/icons/hill.svg',
+          'MOUNTAIN_BIG' => '/icons/mountain.svg',
+        }.freeze
+
+        def price_movement_chart
+          [
+            ['Action', 'Share Price Change'],
+            ['Dividend < 1/2 stock price', '1 ←'],
+            ['Dividend ≥ 1/2 stock price but < stock price', 'none'],
+            ['Dividend ≥ stock price', '1 →'],
+            ['Dividend ≥ 2X stock price', '2 →'],
+            ['Dividend ≥ 3X stock price and stock price ≥ 165', '3 →'],
+            ['Any number of shares sold', '1 ←'],
+            ['Corporation has any shares in the Market at end of an SR', '1 ←'],
+            ['Corporation is sold out at end of an SR', '1 →'],
+          ]
+        end
+
+        def ipo_name(_entity = nil)
+          'Treasury'
+        end
+
+        MOUNTAIN_BIG_HEXES = %w[E21 G21 H22 F26 E27 E29 D30].freeze
+        MOUNTAIN_SMALL_HEXES = %w[G19 H20 E23 D26 D28 G27 H28].freeze
+
+        def setup
+          MOUNTAIN_BIG_HEXES.each { |hex| hex_by_id(hex).assign!('MOUNTAIN_BIG') }
+          MOUNTAIN_SMALL_HEXES.each { |hex| hex_by_id(hex).assign!('MOUNTAIN_SMALL') }
+          corporation_by_id('RB').add_ability(Engine::Ability::Base.new(
+            type: 'FreeTunnel',
+            description: 'Free tunnel'
+          ))
+        end
+
+        def big_mountain?(hex)
+          !hex.assignments.keys.find { |ability| ability.include? :MOUNTAIN_BIG }.nil?
```

Comment:
Don't use `find` unless you need to do something with the value found. And `assignments.keys` will return an array of strings, so you're checking each value in this array to see if `MOUNTAIN_BIG` is a substring.
```suggestion
          hex.assignments.include?('MOUNTAIN_BIG')
```

Reply:
BTW, be careful about labels and strings. Assignments are strings, not labels.